packages/dd-trace/test/plugins/agent.js

```diff
@@ -44,41 +44,43 @@ module.exports = {
       handlers.forEach(handler => handler(req.body))
     })
 
-    return getPort().then(port => {
-      return new Promise((resolve, reject) => {
-        const server = this.server = http.createServer(agent)
-        const emit = server.emit
+    const port = await getPort()
 
-        server.emit = function () {
-          storage.enterWith({ noop: true })
-          return emit.apply(this, arguments)
-        }
+    const server = this.server = http.createServer(agent)
+    const emit = server.emit
 
-        server.on('connection', socket => sockets.push(socket))
-
-        listener = server.listen(port, () => resolve())
+    server.emit = function () {
+      storage.enterWith({ noop: true })
+      return emit.apply(this, arguments)
+    }
 
-        pluginName = [].concat(pluginName)
-        plugins = pluginName
-        config = [].concat(config)
+    server.on('connection', socket => sockets.push(socket))
 
-        server.on('close', () => {
-          tracer = null
-        })
+    const promise = new Promise((resolve, reject) => {
```

Comment:
Wouldn't it be cleaner if we directly returned this `new Promise` instead of define the `const promise` here to use it only in the return of line 83?

In my opinion, it is nothing blocker, so I will accept the PR with this comment

Reply:
Probably. I had thought the ordering mattered, but it might not.